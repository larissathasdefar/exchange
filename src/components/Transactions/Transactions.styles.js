import styled from 'styled-components'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'

export const Container = styled(Paper)`
  max-width: 972px;
  margin: 0 auto;
  margin-bottom: 14px;
  padding: 20px 14px;
  transition: 0.5s;
  opacity: 0.7;

  &:hover {
    opacity: 1;
  }
`

export const BlueText = styled(Typography)`
  && {
    color: #1e88e5;
  }
`
