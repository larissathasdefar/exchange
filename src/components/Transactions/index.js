import React, { PureComponent, Fragment } from 'react'
import _ from 'prop-types'
import Typography from '@material-ui/core/Typography'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableRow from '@material-ui/core/TableRow'
import TableCell from '@material-ui/core/TableCell'
import { Container, BlueText } from './Transactions.styles'
import { format, isToday, isThisYear } from 'date-fns'

const formatMoney = (amount, currency) => amount
  .toLocaleString(undefined, { style: 'currency', currency: currency })

const formatOldDay = date =>
  `${format(date, isThisYear(date) ? 'DD MMM' : 'DD MMM YYYY')} - ${format(date, 'HH:mm')}`

class Transactions extends PureComponent {
  render() {
    const { transactions } = this.props
    return (
      <Container elevation={1}>
        <Typography variant="h5" align="center" paragraph>
          Transactions
        </Typography>
        {
          transactions.length === 0
            ? <Typography align="center">You don't have transactions yet.</Typography>
            : (
              <Table>
                <TableBody>
                  {
                    transactions.map(({ from, to, date }) => (
                      <TableRow key={date.toString()}>
                        <TableCell>
                          <Typography variant="h5">
                            {`Exchanged to ${to.code}`}
                          </Typography>
                          <Typography variant="caption">
                            {
                              isToday(date)
                                ? format(date, 'HH:mm')
                                : formatOldDay(date)
                            }
                          </Typography>
                        </TableCell>
                        <TableCell align="right">
                          <Typography variant="h5">
                            {`- ${formatMoney(from.amount, from.code)}`}
                          </Typography>
                          <BlueText variant="caption">
                            {`+ ${formatMoney(to.amount, to.code)}`}
                          </BlueText>
                        </TableCell>
                      </TableRow>
                    ))
                  }
                </TableBody>
              </Table>
            )
        }
      </Container>
    )
  }
}

Transactions.propTypes = {
  transactions: _.arrayOf(_.shape({
    form: _.shape({
      code: _.string,
      amount: _.number,
    }),
    to: _.shape({
      code: _.string,
      amount: _.number,
    }),
    date: _.date,
  })),
}

export default Transactions
