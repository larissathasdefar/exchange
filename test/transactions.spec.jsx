import React from 'react'
import { shallow } from 'enzyme'

import Transactions from 'components/Transactions'

describe('Transactions', () => {
  it('Having 1 trasaction history', () => {
    const component = shallow(
      <Transactions
        transactions={[
          {
            from: { code: 'EUR', amount: 15 },
            to: { code: 'GBP', amount: 3.19 },
            date: new Date('12/22/2018 08:55'),
          },
        ]}
      />
    )
    expect(component.debug()).toMatchSnapshot()
    expect(component.html()).toContain('Exchanged to GBP')
    expect(component.html()).toContain('22 Dec 2018 - 08:55')
    expect(component.html()).toContain('- € 15,00')
    expect(component.html()).toContain('+ £ 3,19')
  })

  it('Having no trasactions history', () => {
    const component = shallow(
      <Transactions
        transactions={[]}
      />
    )
    expect(component.debug()).toMatchSnapshot()
    expect(component.html()).toContain('You don&#x27;t have transactions yet.')
  })
})