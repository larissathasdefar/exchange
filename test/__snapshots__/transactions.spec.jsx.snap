// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions Having 1 trasaction history 1`] = `
"<Styled(WithStyles(Paper)) elevation={1}>
  <WithStyles(Typography) variant=\\"h5\\" align=\\"center\\" paragraph={true}>
    Transactions
  </WithStyles(Typography)>
  <WithStyles(Table)>
    <WithStyles(TableBody)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)>
          <WithStyles(Typography) variant=\\"h5\\">
            Exchanged to GBP
          </WithStyles(Typography)>
          <WithStyles(Typography) variant=\\"caption\\">
            22 Dec 2018 - 08:55
          </WithStyles(Typography)>
        </WithStyles(TableCell)>
        <WithStyles(TableCell) align=\\"right\\">
          <WithStyles(Typography) variant=\\"h5\\">
            - € 15,00
          </WithStyles(Typography)>
          <Styled(WithStyles(Typography)) variant=\\"caption\\">
            + £ 3,19
          </Styled(WithStyles(Typography))>
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableBody)>
  </WithStyles(Table)>
</Styled(WithStyles(Paper))>"
`;

exports[`Transactions Having no trasactions history 1`] = `
"<Styled(WithStyles(Paper)) elevation={1}>
  <WithStyles(Typography) variant=\\"h5\\" align=\\"center\\" paragraph={true}>
    Transactions
  </WithStyles(Typography)>
  <WithStyles(Typography) align=\\"center\\">
    You don&#39;t have transactions yet.
  </WithStyles(Typography)>
</Styled(WithStyles(Paper))>"
`;
